hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -42,173 +38,241 @@ object HealthTransformations {
-            categorical,
-            conditionOtherDescription = rawRecord.getOptional("hs_dx_infect_other_spec")
-          )
-        // generic case for infectious disease
-        case (disease, categorical) =>
-          createHealthConditionRow(
-            rawRecord,
-            s"dx_${disease}",
-            infectiousDiseaseCondition,
-            categorical
-          )
-      }
-    } else None
-
-  def mapCongenitalEyeDisorder(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
-    if (rawRecord.getBoolean("hs_congenital_yn") && rawRecord.getBoolean("hs_cg_eye_disorders_yn"))
-      eye_cg_disorders.flatMap {
-        // "other" case
-        case ("other", categorical) =>
+  def processHealthCondition(
+    rawRecord: RawRecord,
+    healthCondition: HealthCondition
+  ): Iterable[HlesHealthCondition] = {
+    val congenital: Option[HlesHealthCondition] = if (healthCondition.congenital.isDefined) {
+      // if congenital, then check hs_congenital_yn && healthCondition.congenital._2
+      if (rawRecord.getBoolean("hs_congenital_yn") &&
+          rawRecord.getBoolean(s"hs_${healthCondition.congenital.get._2}_yn"))
+        // congenital other
+        if (conditions.apply(healthCondition.condition) == -1L)
           createHealthConditionRow(
             rawRecord,
-            "cg_eye_other",
-            eyeDiseaseCondition,
-            categorical,
+            healthCondition.congenital.get._1,
+            conditionTypes.apply(healthCondition.conditionType),
+            conditions.apply(healthCondition.condition),
             isCongenital = true,
-            conditionOtherDescription = rawRecord.getOptional("hs_cg_eye_other_spec")
+            conditionOtherDescription =
+              rawRecord.getOptional(s"hs_${healthCondition.congenital.get._1}_spec")
           )
-        // general case
-        case (disorder, categorical) =>
+        // congenital general
+        else if (rawRecord.getBoolean(s"hs_${healthCondition.congenital.get._1}"))
           createHealthConditionRow(
             rawRecord,
-            s"cg_eye_${disorder}",
-            eyeDiseaseCondition,
-            categorical,
+            healthCondition.congenital.get._1,
+            conditionTypes.apply(healthCondition.conditionType),
+            conditions.apply(healthCondition.condition),
             isCongenital = true
           )
-      }
-    else None
-
-  def mapEyeDisease(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
-    if (rawRecord.getBoolean("hs_dx_eye_yn")) {
-      eye_diseases.flatMap {
-        // "other" case
-        case ("eye_other", categorical) =>
+        else None
+      else None
+    } else None
+    val nonCongenital: Option[HlesHealthCondition] = if (healthCondition.nonCongenital.isDefined) {
+      if (rawRecord.getBoolean(s"hs_${healthCondition.nonCongenital.get._2}_yn"))
+        // non-congenital other
+        if (conditions.apply(healthCondition.condition) == -1L)
```

Comment:
I would rather not key off a magic value here. Can we include a boolean in the `HealthCondition` class to drive this instead?

Reply:
ah so like an `isOther` flag? Sure thing! I was toying with that idea anyway, lezdoit